tests/preprocessing/test_warm_up_module.py

```diff
@@ -0,0 +1,200 @@
+    # Temporary path to write pkl to, will be cleaned up at the end of the test.
+    checkpoint_dir = tmp_path.joinpath("resources")
+    checkpoint_dir.mkdir()
+
+    # Save a temporary model using checkpointer
+    saved_model = SmallCnn()
+    checkpointer = BestMetricTorchCheckpointer(str(checkpoint_dir), "best_model.pkl", True)
+    checkpointer.maybe_checkpoint(saved_model, 0.7)
+
+    # Save a temporary weights mapping dict
+    weights_mapping_path = tmp_path.joinpath("data.json")
+    saved_weights_mapping_dict = {
+        "base_module.conv1": "conv1",
+        "base_module.conv2": "conv2",
+        "head_module.fc1": "fc1",
+    }
+    with open(weights_mapping_path, "w") as fp:
+        json.dump(saved_weights_mapping_dict, fp)
+
+    # Load the saved model using warmup module
+    warmup_module = WarmedUpModule(
+        pretrained_model_path=Path(os.path.join(checkpoint_dir, "best_model.pkl")),
+        weights_mapping_path=weights_mapping_path,
+    )
+
+    # Check if the pretrained model state is loaded correctly
+    assert warmup_module.pretrained_model_state is not None
+    for saved_model_value, pretrained_model_value in zip(
+        saved_model.state_dict().values(), warmup_module.pretrained_model_state.values()
+    ):
+        assert (saved_model_value == pretrained_model_value).all()
+
+    # Check if the weights mapping dict is loaded correctly
+    assert warmup_module.weights_mapping_dict is not None
+    for key in warmup_module.weights_mapping_dict.keys():
+        assert warmup_module.weights_mapping_dict[key] == saved_weights_mapping_dict[key]
+
+
+def test_loading_similar_models_without_mapping() -> None:
+    pretrained_model = ToyConvNet()
+    model = ToyConvNet()
+    old_model = copy.deepcopy(model)
+    warmup_module = WarmedUpModule(pretrained_model=pretrained_model)
+    warmup_module.load_from_pretrained(model)
+
+    # Check if the weights are the same with pretrained model after loading
+    for model_value, pretrained_model_value in zip(
+        model.state_dict().values(), pretrained_model.state_dict().values()
+    ):
+        assert (model_value == pretrained_model_value).all()
+
+    # Check if the weights are different from previous model after loading
+    for old_model_value, model_value in zip(old_model.state_dict().values(), model.state_dict().values()):
+        assert (old_model_value != model_value).any()
+
+
+def test_loading_different_models_without_mapping() -> None:
+    pretrained_model = SmallCnn()
+    model = ToyConvNet()
+    old_model = copy.deepcopy(model)
+    warmup_module = WarmedUpModule(pretrained_model=pretrained_model)
+    warmup_module.load_from_pretrained(model)
+
+    # Check if the weights are same with previous model as loading should not have any effect
+    for old_model_value, model_value in zip(old_model.state_dict().values(), model.state_dict().values()):
+        assert (old_model_value == model_value).any()
```

Comment:
I think you want to do

`assert (old_model_value == model_value).all()`

instead.

Reply:
That's true. Thanks for mentioning.